doc/reference/kernel/other/event_manager.rst

```diff
@@ -0,0 +1,396 @@
+.. _event_manager:
+
+Event Manager
+#############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Event Manager is a piece of software that supports development of consistent, modular, event-based application.
+In an event-based application, parts of the application functionality are separated into isolated modules that communicate with each other using events.
+Events are submitted by modules and other modules can subscribe and react to them.
+
+.. figure:: images/em_overview.svg
+   :alt: Architecture of an application based on Event Manager
+
+See the :ref:`event_manager_sample` sample for a simple example of how to use the Event Manager.
+
+Events
+******
+
+Events are structured data types that are defined by the application and can contain additional data.
+
+The Event Manager handles the events by processing and propagating all of them to the modules (listeners) that subscribe to a specific event.
+Multiple modules can subscribe to the same event.
+As part of this communication, listeners can process events differently based on their type.
+
+The Event Manager provides API for defining, creating, and subscribing events.
+See `Implementing an event type`_ for details about how to create custom event types.
+
+.. note::
+   Currently, up to 32 event types can be used in an application.
+
+Modules
+*******
+
+Modules are separate source files that can subscribe to every defined event.
+You can use events for communication between modules.
+
+There is no limitation as to how many modules each module can subscribe to.
+An application can have as many modules as required.
+
+The Event Manager provides API for subscribing modules to a specific events defined in the application.
+When a module is subscribing to a specific event, it is called a listener module.
+Every listener is identified by a unique name.
+
+.. _event_manager_configuration:
+
+Configuration
+*************
+
+To use Event Manager, you must enable the following Kconfig options:
+
+* :kconfig:`CONFIG_EVENT_MANAGER` - This option enables the Event Manager.
+* :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` - This option enables orphan memory sections used by the Event Manager.
```

Comment:
why not fix the orphan sections instead? and, just looking at the code, this Kconfig option seems to be unused and a leftover.

Reply:
Sorry, did not understand your question. If you are asking whether we are using orphan sections, the answer is yes. If you disable this option you wouldn't be able to build sample.